packages/server/tests/acceptance/erc20.spec.ts

```diff
@@ -0,0 +1,295 @@
+            const res = await relay.call('eth_getCode', [contract.address]);
+            expect(res).to.eq(ERC20MockJson.deployedBytecode);
+        });
+
+        describe('set decimals', function () {
+            const decimals = 6;
+
+            it('can set decimals during construction', async function () {
+
+                const decimalsContract = await deployErc20([name, symbol, decimals], ERC20DecimalsMockJson);
+                expect(await decimalsContract.decimals()).to.be.equal(decimals);
+            });
+        });
+
+        describe('should behave like erc20', function() {
+            describe('total supply', function () {
+                it('returns the total amount of tokens', async function () {
+                    const supply = await contract.totalSupply();
+                    expect(supply.toString()).to.be.equal(initialSupply.toString());
+                });
+            });
+
+            describe('balanceOf', function () {
+                describe('when the requested account has no tokens', function () {
+                    it('returns zero', async function () {
+                        const otherBalance = await contract.balanceOf(anotherAccount);
+                        expect(otherBalance.toString()).to.be.equal('0');
+                    });
+                });
+
+                describe('when the requested account has some tokens', function () {
+                    it('returns the total amount of tokens', async function () {
+                        const balance = await contract.balanceOf(initialHolder);
+                        expect(balance.toString()).to.be.equal(initialSupply.toString());
+                    });
+                });
+            });
+
+            describe('transfer from', function () {
+                let spender;
+                let spenderWallet;
+
+                before(async function () {
+                    spender = accounts[1].address;
+                    spenderWallet = accounts[1].wallet;
+                });
+
+                describe('when the token owner is not the zero address', function () {
+                    let tokenOwner, tokenOwnerWallet;
+                    before(async function () {
+                        tokenOwner = initialHolder;
+                        tokenOwnerWallet = accounts[0].wallet;
+                    });
+
+                    describe('when the recipient is not the zero address', function () {
+                        let to, toWallet;
+                        before(async function () {
+                            to = anotherAccount;
+                            toWallet = accounts[2].wallet;
+                        });
+
+                        describe('when the spender has enough allowance', function () {
+                            before(async function () {
+                                await contract.approve(spender, initialSupply, { from: initialHolder });
```

Comment:
Q: should the approve calls be verified?
What's the expected response since this is a feature we provide

Reply:
I believe an event is emitted, you can verify that

Reply:
Contract calls return a transaction result. I don't think we should make a full assertion for that in those tests.

An Approval event is emitted and tested [here](https://github.com/hashgraph/hedera-json-rpc-relay/pull/241/files#diff-31050a1f9515fb6f9d6e5d3b684b0b5aaa81a6e2c7d1c2e7b2552195d79d067dR169)